Numerous fixes and changes

doamatto opened the pull request:
Jack had me look through all the code.

This PR will :
  - Remove false support for Mac PPC and Mac 680x0 from `getOS()`
  - Remove unnecessary imports
  - Replace `~/.openbin/` with XDG specification directories
  - Use symlinks instead of `alias` to create `ob` command
  - Include install scripts as assets in website, replacing reliance on GitHub for downloads
  - Have a Makefile to make building easier
    - Replaced hardcoded SUPABASE_URL and SUPABASE_KEY variables with build-time variables
    - Replaced the old artifact building and deploying script (for GH Releases)
  - Added a check to ensure port 8089 is open for authentication
  - Remove duplicate code and unnecessary branches
  - Replaced (a lot) of logical or with nullish coalescing

Some other things that should be done that I didn't have a chance to do :
  - Rate limiting on the API that is stricter for unauthenticated users
  - Automated tests for the API and CLI
  - Proper documentation and commenting in `web/`
  - Respect user tracking preferences and not load Umami at all if GPC is present
  - Rewriting the footer to support two lines, if you ever add the link list back
  - Add proper buttons to profile page for account management
  - Gracefully handle missing environment variables at server start
  - Gracefully handle missing SUPABASE_SERVICE_KEY on account deletion
  - Fix non-functional date picker
  - Gracefully handle disallowed code publishing (web)
  - Consider using username (from GH or whatever) instead of UUID for profile URLs

doamatto committed:
chore: numerous fixes and changes

Jack had me look through all the code. My laptop is haha-funny some times, so it was
smarter for me to just use one commit – apologies for that.

This commit :
  - Removed false support for Mac PPC and Mac 680x0 from `getOS()`
  - Removed unnecessary imports
  - Replace `~/.openbin/` with XDG specification directories
  - Use symlinks instead of `alias` to create `ob` command
  - Include install scripts as assets in website, replacing reliance on GitHub for downloads
  - Made Makefile to make building easier
    - Replaced hardcoded SUPABASE_URL and SUPABASE_KEY variables with build-time variables
  - Added a check to ensure port is open for authentication
  - Remove duplicate code and unnecessary branches
  - Replaced (a lot) of logical or (`||`) to nullish coalescing
  - Wrote some basic tests for the Golang CLI

Some other things that should be done that I didn't have a chance to do :
  - Rate limiting on the API that is stricter for unauthenticated users
  - Tests for the API
  - Proper documentation and commenting in `web/`
  - Respect user tracking preferences and not load Umami at all if GPC is present
  - Rewriting the footer to support two lines, if you ever add the link list back
  - Add proper buttons to profile page for account management
  - Gracefully handle missing environment variables at server start
  - Gracefully handle missing SUPA_ABSE_SERVICE_KEY on account deletion
  - Fix non-functional date picker
  - Gracefully handle disallowed code publishing (web)
  - Consider using username (from GH or whatever) instead of UUID for profile URLs

doamatto committed:
fix: remove unnecessary import

doamatto committed:
fix: finish writing publish.yml

doamatto committed:
chore: remove legacy binary tool

doamatto committed:
fix: add missing GITHUB_TOKEN

Turns out I still have to write that. It just generates automatically now.

lavylavenderr commented:
Woah! This is amazing! We'll get back to your PR when make a seperate branch so the judges over at Supabase know what to look at.

doamatto committed:
fix: SUPABASE_KEY and SUPERBASE_URL not being properly pointed

This also has a bonus effect of now automatically opening the browser
so a user can authenticate if it's on a common supported OS.

lavylavenderr commented:
Hey there @doamatto, we've decided against merging any commits until after the winners and results are released by Supabase, so we'll loop back to this in a little bit. Thanks!

tessamerrill commented on web/src/app/install.sh/route.ts:
> Include install scripts as assets in website, replacing reliance on GitHub for downloads

This is kinda pointless, as the scripts download from GitHub Releases. So if GitHub goes down they wouldn't be able to download no matter what.

doamatto replied:
Sure, totally, except there's no good reason to redirect just for downloading the script.

On a completely valid note, it wouldn't *technically* be. Git tags use the "Git Operations" on the status page. GitHub Releases are based on Git tags, but are part of "API requests" iirc. So, in theory, it would mean that the script can go down and the binary downloads still be functional in its current state. This would require the API to also go down. That said, GitHub in general is.. flaky at best.

ethndotsh replied:
Thanks for the feedback. We will take your theoretical and improbable situation for a non-critical part of our infrastructure into mind 🥰🥰

tessamerrill replied:
It would be better for the script to fail from the get-go instead of failing while downloading the binary. Hosting the binaries would also be too much work.

ethndotsh replied:
However we are incredibly thankful for your contributions.

doamatto committed:
chore: revert changes for route.ts scripts
